rocketpy/mathutils/function.py

```diff
@@ -13,13 +13,22 @@
 import matplotlib.pyplot as plt
 import numpy as np
 from scipy import integrate, linalg, optimize
+from bisect import bisect_left
 
 try:
     from functools import cached_property
 except ImportError:
     from ..tools import cached_property
 
 NUMERICAL_TYPES = (float, int, complex, np.ndarray, np.integer, np.floating)
+INTERPOLATION_TYPES = {
+    "linear": 0,
+    "polynomial": 1,
+    "akima": 2,
+    "spline": 3,
+    "shepard": 4,
+}
+EXTRAPOLATION_TYPES = {"zero": 0, "natural": 1, "constant": 2}
```

Comment:
When you asked before I said I preferred these dictionaries to be save within the Function class since they will not (probably) be used in other contexts.

HOWEVER, now I think it's a good idea to do what you did here, because these constants will be defined only once in te execution time. If we save them as class attributes, they would be defined again and again every time we initiate a Function.